torch/custom_class.h

```diff
@@ -18,6 +18,32 @@
 
 namespace torch {
 
+/// This struct is used to represent default values for arguments
+/// when registering methods for custom classes.
+///     static auto register_foo = torch::class_<Foo>("myclasses", "Foo")
+///       .def("myMethod", &Foo::myMethod, {torch::arg("name") = name});
+struct arg {
+  // Explicit constructor.
+  explicit arg(std::string name) : name_(std::move(name)), value_set_(false) {}
+  // Assignment operator. This enables the pybind-like syntax of
+  // torch::arg("name") = value.
+  arg& operator=(const c10::IValue& rhs) {
+    value_ = rhs;
+    value_set_ = true;
+    return *this;
+  }
+
+  // The name of the argument. This is copied to the schema; argument
+  // names cannot be extracted from the C++ declaration.
+  std::string name_;
+  // IValue's default constructor makes it None, which not distinguishable from
```

Comment:
s/which not/which is not/

Could we have a torch::arg::None method that boils down to an empty IValue? This would make the API more intuitive and resembles Pybind more.